kpet/run.py

```diff
@@ -62,31 +62,77 @@ def __init__(self, suite, cases):
         self.cases = cases
 
 
+class Test:
+    # pylint: disable=too-few-public-methods, too-many-instance-attributes
+    """A test run - an execution of a particular case of a test suite"""
+
+    def __init__(self, suite, case):
+        """
+        Initialize a test run.
+
+        Args:
+            suite:  The test suite (kpet.data.Suite) being executed.
+            case:   The test case of the test suite (kpet.data.Case) being
+                    executed.
+        """
+        assert isinstance(suite, data.Suite)
+        assert isinstance(case, data.Case)
+        assert case in suite.cases
+
+        # pylint: disable=invalid-name
+        # TODO Remove conditions once IDs are mandatory
+        self.id = (suite.id or "") + "." + (case.id or "")
+        self.name = " - ".join(filter(None, (suite.name, case.name)))
+        self.origin = suite.origin
+        self.location = suite.location
+        self.max_duration_seconds = case.max_duration_seconds
+        self.hostRequires = filter(None,
+                                   (suite.hostRequires, case.hostRequires))
+        self.partitions = filter(None,
+                                 (suite.partitions, case.partitions))
+        self.kickstart = filter(None,
+                                (suite.kickstart, case.kickstart))
+        self.waived = \
+            case.waived if case.waived is not None else \
+            suite.waived if suite.waived is not None else \
+            False
```

Comment:
They can be `True`, `False` or `None`. Where `None` means absent, or "no effect". The lower level specification is supposed to override the higher level if it's not `None`. The compact way you suggest won't work because then a particular case of a waived suite cannot be unwaived.